src/c++/perf_analyzer/genai-perf/tests/test_csv_exporter.py

```diff
@@ -66,102 +62,149 @@ def write(self: Any, content: str) -> int:
 
         return written_data
 
-    def test_csv_output(self, mock_read_write: pytest.MonkeyPatch) -> None:
+    def test_streaming_llm_csv_output(
+        self, monkeypatch, mock_read_write: pytest.MonkeyPatch
+    ) -> None:
         """
         Collect LLM metrics from profile export data and confirm correct values are
         printed in csv.
         """
-
-        tokenizer = get_tokenizer(DEFAULT_TOKENIZER)
-        pd = LLMProfileDataParser(
-            filename=Path("triton_profile_export.json"),
-            tokenizer=tokenizer,
+        argv = [
+            "genai-perf",
+            "-m",
+            "model_name",
+            "--service-kind",
+            "openai",
+            "--endpoint-type",
+            "chat",
+            "--streaming",
+        ]
+        monkeypatch.setattr("sys.argv", argv)
+        args, _ = parser.parse_args()
+
+        metrics = LLMMetrics(
+            request_throughputs=[123],
+            request_latencies=[4, 5, 6],
+            time_to_first_tokens=[7, 8, 9],
+            inter_token_latencies=[10, 11, 12],
+            output_token_throughputs=[456],
+            output_sequence_lengths=[1, 2, 3],
+            input_sequence_lengths=[5, 6, 7],
         )
-        stat = pd.get_statistics(infer_mode="concurrency", load_level="10")
+        stats = Statistics(metrics=metrics)
+
+        config = ExporterConfig()
+        config.stats = stats.stats_dict
+        config.metrics = stats.metrics
+        config.artifact_dir = Path(".")
+        config.args = args
+
+        exporter = CsvExporter(config)
+        exporter.export()
 
         expected_content = [
             "Metric,avg,min,max,p99,p95,p90,p75,p50,p25\r\n",
-            "Time To First Token (ms),2.00,2.00,2.00,2.00,2.00,2.00,2.00,2.00,2.00\r\n",
-            "Inter Token Latency (ms),1.50,1.00,2.00,1.99,1.95,1.90,1.75,1.50,1.25\r\n",
-            "Request Latency (ms),8.00,7.00,9.00,8.98,8.90,8.80,8.50,8.00,7.50\r\n",
-            "Output Sequence Length,4.50,3.00,6.00,5.97,5.85,5.70,5.25,4.50,3.75\r\n",
-            "Input Sequence Length,3.50,3.00,4.00,3.99,3.95,3.90,3.75,3.50,3.25\r\n",
+            "Request Latency (ms),5.00,4.00,6.00,5.98,5.90,5.80,5.50,5.00,4.50\r\n",
```

Comment:
The order changing has the potential to be breaking even more than the console.
The json output has keys we can leverage to avoid issues. Parsing the CSV has a greater potential to rely on which row contains the data.

Reply:
I agree this is a breaking change but I feel like the change is needed. Previously, there was no good order imposed on these metrics outputs, but now we do. We go from base metrics to task specific metrics, and we will need this ordering as we add more features and support other services.

For instance, if we look at the previous LLM csv output
```
Metric,avg,min,max,p99,p95,p90,p75,p50,p25,
Time To First Token (ms),2.00,2.00,2.00,2.00,2.00,2.00,2.00,2.00,2.00,
Inter Token Latency (ms),1.50,1.00,2.00,1.99,1.95,1.90,1.75,1.50,1.25,
Request Latency (ms),8.00,7.00,9.00,8.98,8.90,8.80,8.50,8.00,7.50,
Output Sequence Length,4.50,3.00,6.00,5.97,5.85,5.70,5.25,4.50,3.75,
Input Sequence Length,3.50,3.00,4.00,3.99,3.95,3.90,3.75,3.50,3.25,

Metric,Value,
Output Token Throughput (per sec),900000000.00,
Request Throughput (per sec),200000000.00,
```
and embedding csv output
```
Metric,avg,min,max,p99,p95,p90,p75,p50,p25,
Request Latency (ms),8.00,7.00,9.00,8.98,8.90,8.80,8.50,8.00,7.50,

Metric,Value,
Request Throughput (per sec),200000000.00,
```
we see that request latency is in the middle of the request level metrics for LLM and request throughput is below the output token throughput. Instead if both base metrics were located on the top, it will be consistent with embeddings csv output and users consuming the file has more consistent pattern. This is same for other future tasks we will add, giving the users patterns/orders like
```
Metric,avg,min,max,p99,p95,p90,p75,p50,p25,
Request Latency (ms),8.00,7.00,9.00,8.98,8.90,8.80,8.50,8.00,7.50,  # base metric
... task specific metrics

Metric,Value,
Request Throughput (per sec),200000000.00,  # base metric
... task specific metrics
```

Reply:
You are right. 
Its just a separation of the refactor vs the clean up that we are now shooting for in our PRs

Reply:
Same as above comment